vortex-layout/src/layouts/flat/writer.rs

```diff
@@ -49,11 +49,14 @@ impl FlatLayoutWriter {
     }
 }
 
-fn retain_only_stats(array: &dyn Array, stats: &[Stat]) {
+fn update_stats(array: &dyn Array, stats: &[Stat]) -> VortexResult<()> {
```

Comment:
Why the name change? I think since we removed the big stats this method does nothing since we always pass all stats. We could delete it and add it back in the future